packages/w3p/src/providers/wrapped/evm-wallet-connect.ts

```diff
@@ -0,0 +1,316 @@
+            'wallet_switchEthereumChain',
+            'wallet_addEthereumChain',
+            'eth_requestAccounts',
+
+            'eth_sendTransaction',
+
+            'eth_sign',
+            'personal_sign',
+            'eth_signTypedData',
+          ],
+          chains: ['eip155:1'],
+          events: [
+            PROVIDER_EVENTS.Connect,
+            PROVIDER_EVENTS.Disconnect,
+            PROVIDER_EVENTS.ChainChanged,
+            PROVIDER_EVENTS.AccountsChanged,
+          ],
+        },
+      },
+      optionalNamespaces: {
+        eip155: {
+          methods: [
+            'wallet_switchEthereumChain',
+            'wallet_addEthereumChain',
+            'eth_requestAccounts',
+
+            'eth_sendTransaction',
+
+            'eth_sign',
+            'personal_sign',
+            'eth_signTypedData',
+          ],
+          chains: ['eip155:1'],
+          events: [
+            PROVIDER_EVENTS.Connect,
+            PROVIDER_EVENTS.Disconnect,
+            PROVIDER_EVENTS.ChainChanged,
+            PROVIDER_EVENTS.AccountsChanged,
+          ],
+        },
+      },
+      skipPairing: true,
+    })
+
+    await this.#rawProvider.enable()
+
+    this.#ethProvider = new providers.Web3Provider(this.#rawProvider)
+
+    const accounts = await this.#rawProvider.request<string[]>({
+      method: 'eth_requestAccounts',
+    })
+
+    this.#chainId =
+      this.#rawProvider?.session?.namespaces?.eip155?.chains?.[0]?.split(':')[1]
+
+    this.#address = accounts?.[0]
+
+    this.emit(
+      PROVIDER_EVENT_BUS_EVENTS.AccountChanged,
+      this.#defaultEventPayload,
+    )
+
+    this.emit(
+      this.isConnected
+        ? PROVIDER_EVENT_BUS_EVENTS.Connect
+        : PROVIDER_EVENT_BUS_EVENTS.Disconnect,
+      this.#defaultEventPayload,
+    )
+
+    this.#w3Modal.closeModal()
+  }
+
+  async disconnect() {
+    await this.#rawProvider.disconnect()
+    this.#chainId = undefined
+    this.#address = undefined
+    this.emit(PROVIDER_EVENT_BUS_EVENTS.Disconnect, this.#defaultEventPayload)
+  }
+
+  async #checkForPersistedSession() {
+    if (!this.#rawProvider) {
+      throw new ReferenceError('EthereumProvider is not initialized.')
+    }
+
+    if (!this.#rawProvider?.session) return
+
+    this.#chainId =
+      this.#rawProvider?.session?.namespaces?.eip155?.chains?.[0]?.split(':')[1]
+
+    this.#address =
+      this.#rawProvider?.session?.namespaces?.eip155?.accounts?.[0]?.split(
+        ':',
+      )?.[2]
+  }
+
+  getAddressUrl(chain: Chain, address: string): string {
+    return getEthExplorerAddressUrl(chain, address)
+  }
+
+  getTxUrl(chain: Chain, txHash: string): string {
+    return getEthExplorerTxUrl(chain, txHash)
+  }
+
+  getHashFromTx(txResponse: TransactionResponse): string {
+    return (txResponse as EthTransactionResponse).transactionHash
+  }
+
+  /**
+   * @description Switch the chain with WalletConnect Provider.
+   * You should always add a network first and then change it.
```

Comment:
add example

Reply:
u should only pass the chain id to this method, it will work like in any another provider

Reply:
then fix the description like this:
/**
 * @description Switch the chain with WalletConnect Provider.
 * You should always add a network first with the proper method {@link Provider.setChainDetails} and then change it.
 */